### PR TITLE
Minor changes to release docs for clarity.

### PR DESCRIPTION
**Pull request by tellthemachines**
## Description

Adding a few clarifications to the release docs, particularly around WordPress release related tasks.

### CODE REVIEW

File: docs/contributors/release.md [hunk 1]
```diff
@@ -311,7 +312,7 @@ Now _cherry-pick_ the commits from `master` to `wp/trunk`, use `-m 1 commithash`
 1. `git cherry-pick -m 1 cb150a2`
 2. `git push`
 
-Whilst waiting for the Travis CI build for  `wp/trunk` [branch to pass](https://travis-ci.com/WordPress/gutenberg/branches) identify and begin updating the `CHANGELOG.md` files:
+Whilst waiting for the build for `wp/trunk`[branch to pass](https://travis-ci.com/WordPress/gutenberg/branches), identify and begin updating the `CHANGELOG.md` files:
```
Comment (gziolo) [inline docs/contributors/release.md:315]: It isn't Travis anymore but GitHub actions now 😃

**Review by gziolo — APPROVED** [review]
Thank you for proposing those revisions 👍

File: docs/contributors/release.md (hunk 1 above)
Comment (tellthemachines) [reply docs/contributors/release.md:315]: Yeah, I'm not sure what the new link for that would be though. It's not obvious from the GH actions interface.

Comment (talldan) [reply docs/contributors/release.md:315]: Maybe https://github.com/WordPress/gutenberg/actions. Or could remove the link, the old one wasn't that useful as it showed all branches.

**Pull request merged**